### DISCUSSION TITLE
Choose the user agent in request header

### DISCUSSION THREAD

**Discussion by MinatSukiBotNET** [FlareSolverr/FlareSolverr#1098] [Ideas]
Would it be possible to use our own user agent for flaresolver requests and not use the default user agent, e.g. if the user agent is not defined, leave the default flaresolver UA, otherwise use it?

**Comment by demon101** [comment]
Have you found a solution?

**Comment by svaningelgem** [comment]
I also wanted this, and made something that worked:
- https://github.com/svaningelgem/FlareSolverr/tree/set_useragent <-- based on the nodriver image I made
- https://github.com/svaningelgem/FlareSolverr/tree/allow-to-set-useragent-in-session-creation <-- based on the current master branch here

You can pass `userAgent` any time you want. But watch out. CF ***will*** detect any modification to it, and you will need to re-authenticate.

I use this myself in order to set a custom user agent string during the whole session with the end-website.

**Comment by ilike2burnthing** [comment]
#266
